src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -242,6 +242,15 @@ export class BundleDataClient {
         })
       )
     );
+    Object.values(data.bundleFillsV3).forEach((x) =>
+      Object.values(x).forEach((fills) =>
+        fills.fills.forEach((fill) => {
+          if (fill.messageHash === UNDEFINED_MESSAGE_HASH) {
+            fill.messageHash = getMessageHash(fill.message);
+          }
+        })
+      )
+    );
```

Comment:
This is tbh clunky and annoying, but it should also be temporary. Once we pass the 7 day deprecation hurdle we can start to strip out the legacy events and can require that the `messageHash` field is present in arweave data retrieved. This would also involve a version bump.

Reply:
I'd like to add a unit test for this because it'll be hard to verify in prod if something goes wrong

Reply:
I added tests for this and `BundleFillsV3` in the relayer in anticipation of the sdk bump https://github.com/UMAprotocol/across-configs/pull/34.

Reply:
I also have some here: https://github.com/across-protocol/sdk/pull/850/commits/3b683328a1004e5336972006dbe6c08e2d4a34cd